structuredheadings/plugin.js

```diff
@@ -649,7 +602,66 @@
           }
         }
       },
+      applyHeadingPreset: {
+        clearAutonumberClassesForHeading: function (editor, heading) {
+          clearStyles(editor, heading);
+          cssUtils.clearLevel(editor, heading);
+          cssUtils.clearNumbering(editor, heading);
+        },
 
+        setAutonumberClassesForHeading: function (editor, value, heading) {
+          cssUtils.setNumbering(editor, heading);
+          cssUtils.setLevel(editor, heading);
+          setCurrentStyle(editor, heading, value);
+        },
+
+        handleCollapsedSelection: function (editor) {
+          // avoid an extra snapshot created by the matchHeading command
+          editor.fire("lockSnapshot", { dontUpdate: true });
+          editor.execCommand("matchHeading");
+          editor.fire("unlockSnapshot");
+
+          // return the one matched current heading
+
+          return [CKEDITOR.plugins.structuredheadings.getCurrentBlockFromPath(editor)];
+        },
+
+        exec: function (editor, value) {
+          var selection = editor.getSelection();
+          var headings = CKEDITOR.plugins.structuredheadings.getHeadingsInSelection(
+                editor,
+                selection
+              );
+          var func;
+
+          if (value === "clear") {
+            func = this.clearAutonumberClassesForHeading.bind(this, editor);
+          } else {
+            func = this.setAutonumberClassesForHeading.bind(this, editor, value);
```

Comment:
the bind params were changed since the signatures changed when moved from dialog to command